.github/workflows/build-test-deploy.yml

```diff
@@ -81,33 +88,34 @@ jobs:
         run: mvn test --batch-mode -Dskip.npm
 
   build-docker-image:
-      needs: [build]
-      runs-on: ubuntu-latest
-      timeout-minutes: 30
-      env:
-        BUILD_NUMBER: ${{ needs.build.outputs.build_number }}
-        DOCKER_IMAGE_TAG: ${{ needs.build.outputs.docker_image_tag }}
-        GIT_COMMIT_HASH: ${{ github.sha }}
-      steps:
-        - uses: actions/checkout@v3
-
-        - name: Build Docker image
-          run: |
-            docker build --tag ${DOCKER_IMAGE_TAG} \
-            --build-arg versienummer=${BUILD_NUMBER} \
-            --build-arg buildId=${BUILD_NUMBER} \
-            --build-arg commit=${GIT_COMMIT_HASH} \
-            --file Containerfile \
-            .
-
-        - name: Save Docker Image
-          run: docker save --output docker-image.tar ${DOCKER_IMAGE_TAG}
-
-        - name: Cache Docker Image
-          uses: actions/cache/save@v3
-          with:
-            path: docker-image.tar
-            key: docker-image-${{ github.repository }}-${{ github.ref_name }}-${{ github.run_number }}
+    needs: [build]
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    env:
+      BUILD_NUMBER: ${{ needs.build.outputs.build_number }}
+      DOCKER_IMAGE_TAG: ${{ needs.build.outputs.docker_image_tag }}
+      DOCKER_IMAGE_LATEST_TAG: ${{ needs.build.outputs.docker_image_latest_tag }}
+      GIT_COMMIT_HASH: ${{ github.sha }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Build Docker image
+        run: |
+          docker build --tag ${DOCKER_IMAGE_TAG} ${{ contains('refs/heads/main', github.ref) && '--tag ${DOCKER_IMAGE_LATEST_TAG} ' }}\
```

Comment:
Maybe add a little comment before this line to explain what is going on and why we are doing this? It's a bit hard to understand. ;-) so indeed, we are still using the unique ${DOCKER_IMAGE_TAG}  but then we are also setting the 'latest' tag but only when on the main branch? I think that is fine indeed.